### ISSUE TITLE
E3022 false positives?

### ISSUE THREAD

**Issue by krisb78** [aws-cloudformation/cfn-python-lint#1847]
*cfn-lint version: 0.44.0*

*False-positive E3022 errors being raised when multiple templates are analysed in one go*

I have a couple of templates that are creating VPCs, each with a number of subnets in it.
All templates had been passing all the checks until version 0.44.0. When we upgraded from 0.43.0, we started seeing:

```

E3022 SubnetId in PublicSubnetAzBRtbAssociation is also associated with PublicSubnetAzBSubnetRtbAssociation
--
827 | ./cloudformation/prod/gmg-prod-vpc.yaml:62:1
828 |  
829 | E3022 SubnetId in PrivateSubnetAzBRtbAssociation is also associated with PrivateSubnetAzBSubnetRtbAssociation
830 | ./cloudformation/prod/gmg-prod-vpc.yaml:62:1
831 |  
832 | E3022 SubnetId in PublicSubnetAzBSubnetRtbAssociation is also associated with PublicSubnetAzBRtbAssociation
833 | ./cloudformation/prod/gmg-prod-vpc.yaml:434:7
834 |  
835 | E3022 SubnetId in PrivateSubnetAzBSubnetRtbAssociation is also associated with PrivateSubnetAzBRtbAssociation
836 | ./cloudformation/prod/gmg-prod-vpc.yaml:444:7
837 |  
838 | E3022 SubnetId in PublicSubnetAzBRtbAssociation is also associated with PublicSubnetAzBSubnetRtbAssociation
839 | ./cloudformation/staging/gmg-staging-common.yaml:5:1
840 |  
841 | E3022 SubnetId in PrivateSubnetAzBRtbAssociation is also associated with PrivateSubnetAzBSubnetRtbAssociation
842 | ./cloudformation/staging/gmg-staging-common.yaml:5:1
843 |  
844 | E3022 SubnetId in PublicSubnetAzBSubnetRtbAssociation is also associated with PublicSubnetAzBRtbAssociation
845 | ./cloudformation/staging/gmg-staging-common.yaml:5:1
846 |  
847 | E3022 SubnetId in PrivateSubnetAzBSubnetRtbAssociation is also associated with PrivateSubnetAzBRtbAssociation
848 | ./cloudformation/staging/gmg-staging-common.yaml:5:1
849 |  
850 | E3022 SubnetId in PublicSubnetAzBRtbAssociation is also associated with PublicSubnetAzBSubnetRtbAssociation
851 | ./cloudformation/staging/gmg-staging-docker.yaml:21:1
852 |  
853 | E3022 SubnetId in PrivateSubnetAzBRtbAssociation is also associated with PrivateSubnetAzBSubnetRtbAssociation
854 | ./cloudformation/staging/gmg-staging-docker.yaml:21:1
855 |  
856 | E3022 SubnetId in PublicSubnetAzBSubnetRtbAssociation is also associated with PublicSubnetAzBRtbAssociation
857 | ./cloudformation/staging/gmg-staging-docker.yaml:21:1
858 |  
859 | E3022 SubnetId in PrivateSubnetAzBSubnetRtbAssociation is also associated with PrivateSubnetAzBRtbAssociation
860 | ./cloudformation/staging/gmg-staging-docker.yaml:21:1
861 |  
862 | E3022 SubnetId in PublicSubnetAzBRtbAssociation is also associated with PublicSubnetAzBSubnetRtbAssociation
863 | ./cloudformation/staging/gmg-staging-dynamo.yaml:9:1
864 |  
865 | E3022 SubnetId in PrivateSubnetAzBRtbAssociation is also associated with PrivateSubnetAzBSubnetRtbAssociation
866 | ./cloudformation/staging/gmg-staging-dynamo.yaml:9:1
867 |  
868 | E3022 SubnetId in PublicSubnetAzBSubnetRtbAssociation is also associated with PublicSubnetAzBRtbAssociation
869 | ./cloudformation/staging/gmg-staging-dynamo.yaml:9:1
870 |  
871 | E3022 SubnetId in PrivateSubnetAzBSubnetRtbAssociation is also associated with PrivateSubnetAzBRtbAssociation
872 | ./cloudformation/staging/gmg-staging-dynamo.yaml:9:1
873 |  
874 | E3022 SubnetId in PublicSubnetAzBRtbAssociation is also associated with PublicSubnetAzBSubnetRtbAssociation
875 | ./cloudformation/staging/gmg-staging-episerver.yaml:92:1
876 |  
877 | E3022 SubnetId in PrivateSubnetAzBRtbAssociation is also associated with PrivateSubnetAzBSubnetRtbAssociation
878 | ./cloudformation/staging/gmg-staging-episerver.yaml:92:1
879 |  
880 | E3022 SubnetId in PublicSubnetAzBSubnetRtbAssociation is also associated with PublicSubnetAzBRtbAssociation
881 | ./cloudformation/staging/gmg-staging-episerver.yaml:92:1
882 |  
883 | E3022 SubnetId in PrivateSubnetAzBSubnetRtbAssociation is also associated with PrivateSubnetAzBRtbAssociation
884 | ./cloudformation/staging/gmg-staging-episerver.yaml:92:1
885 |  
886 | E3022 SubnetId in PublicSubnetAzBRtbAssociation is also associated with PublicSubnetAzBSubnetRtbAssociation
887 | ./cloudformation/staging/gmg-staging-horus.yaml:2:1
888 |  
889 | E3022 SubnetId in PrivateSubnetAzBRtbAssociation is also associated with PrivateSubnetAzBSubnetRtbAssociation
890 | ./cloudformation/staging/gmg-staging-horus.yaml:2:1
891 |  
892 | E3022 SubnetId in PublicSubnetAzBSubnetRtbAssociation is also associated with PublicSubnetAzBRtbAssociation
893 | ./cloudformation/staging/gmg-staging-horus.yaml:2:1
894 |  
895 | E3022 SubnetId in PrivateSubnetAzBSubnetRtbAssociation is also associated with PrivateSubnetAzBRtbAssociation
896 | ./cloudformation/staging/gmg-staging-horus.yaml:2:1
897 |  
898 | E3022 SubnetId in PublicSubnetAzBRtbAssociation is also associated with PublicSubnetAzBSubnetRtbAssociation
899 | ./cloudformation/staging/gmg-staging-keymovement-stream.yaml:9:1
900 |  
901 | E3022 SubnetId in PrivateSubnetAzBRtbAssociation is also associated with PrivateSubnetAzBSubnetRtbAssociation
902 | ./cloudformation/staging/gmg-staging-keymovement-stream.yaml:9:1
903 |  
904 | E3022 SubnetId in PublicSubnetAzBSubnetRtbAssociation is also associated with PublicSubnetAzBRtbAssociation
905 | ./cloudformation/staging/gmg-staging-keymovement-stream.yaml:9:1
906 |  
907 | E3022 SubnetId in PrivateSubnetAzBSubnetRtbAssociation is also associated with PrivateSubnetAzBRtbAssociation
908 | ./cloudformation/staging/gmg-staging-keymovement-stream.yaml:9:1
909 |  
910 | E3022 SubnetId in PublicSubnetAzBRtbAssociation is also associated with PublicSubnetAzBSubnetRtbAssociation
911 | ./cloudformation/staging/gmg-staging-legacy.yaml:25:1
912 |  
913 | E3022 SubnetId in PrivateSubnetAzBRtbAssociation is also associated with PrivateSubnetAzBSubnetRtbAssociation
914 | ./cloudformation/staging/gmg-staging-legacy.yaml:25:1
915 |  
916 | E3022 SubnetId in PublicSubnetAzBSubnetRtbAssociation is also associated with PublicSubnetAzBRtbAssociation
917 | ./cloudformation/staging/gmg-staging-legacy.yaml:25:1
918 |  
919 | E3022 SubnetId in PrivateSubnetAzBSubnetRtbAssociation is also associated with PrivateSubnetAzBRtbAssociation
920 | ./cloudformation/staging/gmg-staging-legacy.yaml:25:1
921 |  
922 | E3022 SubnetId in PublicSubnetAzBRtbAssociation is also associated with PublicSubnetAzBSubnetRtbAssociation
923 | ./cloudformation/staging/gmg-staging-oldshop-services.yaml:9:1
924 |  
925 | E3022 SubnetId in PrivateSubnetAzBRtbAssociation is also associated with PrivateSubnetAzBSubnetRtbAssociation
926 | ./cloudformation/staging/gmg-staging-oldshop-services.yaml:9:1
927 |  
928 | E3022 SubnetId in PublicSubnetAzBSubnetRtbAssociation is also associated with PublicSubnetAzBRtbAssociation
929 | ./cloudformation/staging/gmg-staging-oldshop-services.yaml:9:1
930 |  
931 | E3022 SubnetId in PrivateSubnetAzBSubnetRtbAssociation is also associated with PrivateSubnetAzBRtbAssociation
932 | ./cloudformation/staging/gmg-staging-oldshop-services.yaml:9:1
933 |  
934 | E3022 SubnetId in PublicSubnetAzBRtbAssociation is also associated with PublicSubnetAzBSubnetRtbAssociation
935 | ./cloudformation/staging/gmg-staging-reporting.yaml:19:1
936 |  
937 | E3022 SubnetId in PrivateSubnetAzBRtbAssociation is also associated with PrivateSubnetAzBSubnetRtbAssociation
938 | ./cloudformation/staging/gmg-staging-reporting.yaml:19:1
939 |  
940 | E3022 SubnetId in PublicSubnetAzBSubnetRtbAssociation is also associated with PublicSubnetAzBRtbAssociation
941 | ./cloudformation/staging/gmg-staging-reporting.yaml:19:1
942 |  
943 | E3022 SubnetId in PrivateSubnetAzBSubnetRtbAssociation is also associated with PrivateSubnetAzBRtbAssociation
944 | ./cloudformation/staging/gmg-staging-reporting.yaml:19:1
945 |  
946 | E3022 SubnetId in PublicSubnetAzBRtbAssociation is also associated with PublicSubnetAzBSubnetRtbAssociation
947 | ./cloudformation/staging/gmg-staging-security-groups.yaml:2:1
948 |  
949 | E3022 SubnetId in PrivateSubnetAzBRtbAssociation is also associated with PrivateSubnetAzBSubnetRtbAssociation
950 | ./cloudformation/staging/gmg-staging-security-groups.yaml:2:1
951 |  
952 | E3022 SubnetId in PublicSubnetAzBSubnetRtbAssociation is also associated with PublicSubnetAzBRtbAssociation
953 | ./cloudformation/staging/gmg-staging-security-groups.yaml:2:1
954 |  
955 | E3022 SubnetId in PrivateSubnetAzBSubnetRtbAssociation is also associated with PrivateSubnetAzBRtbAssociation
956 | ./cloudformation/staging/gmg-staging-security-groups.yaml:2:1
957 |  
958 | E3022 SubnetId in PublicSubnetAzBSubnetRtbAssociation is also associated with PublicSubnetAzBRtbAssociation, PublicSubnetAzBRtbAssociation
959 | ./cloudformation/staging/gmg-staging-vpc.yaml:49:1
960 |  
961 | E3022 SubnetId in PrivateSubnetAzBSubnetRtbAssociation is also associated with PrivateSubnetAzBRtbAssociation, PrivateSubnetAzBRtbAssociation
962 | ./cloudformation/staging/gmg-staging-vpc.yaml:49:1
963 |  
964 | E3022 SubnetId in PublicSubnetAzBRtbAssociation is also associated with PublicSubnetAzBSubnetRtbAssociation
965 | ./cloudformation/staging/gmg-staging-vpc.yaml:382:7
966 |  
967 | E3022 SubnetId in PrivateSubnetAzBRtbAssociation is also associated with PrivateSubnetAzBSubnetRtbAssociation
968 | ./cloudformation/staging/gmg-staging-vpc.yaml:392:7
969 |  
970 | E3022 SubnetId in PublicSubnetAzBRtbAssociation is also associated with PublicSubnetAzBSubnetRtbAssociation
971 | ./cloudformation/staging/main.yaml:15:1
972 |  
973 | E3022 SubnetId in PrivateSubnetAzBRtbAssociation is also associated with PrivateSubnetAzBSubnetRtbAssociation
974 | ./cloudformation/staging/main.yaml:15:1
975 |  
976 | E3022 SubnetId in PublicSubnetAzBSubnetRtbAssociation is also associated with PublicSubnetAzBRtbAssociation, PublicSubnetAzBRtbAssociation
977 | ./cloudformation/staging/main.yaml:15:1
978 |  
979 | E3022 SubnetId in PrivateSubnetAzBSubnetRtbAssociation is also associated with PrivateSubnetAzBRtbAssociation, PrivateSubnetAzBRtbAssociation
980 | ./cloudformation/staging/main.yaml:15:1

``` 

Interestingly, PublicSubnetAzBRtbAssociation does not even exist as a logical id in gmg-prod-vpc.yaml, but it does exist in gmg-staging-vpc.yaml and gmg-mgmnt-vpc.yaml.

Each of these templates creates a set of subnet ids that do share the same logical ids, but since they are used to create spearate stack, I don't think this should be an issue.

Can it be related to https://github.com/aws-cloudformation/cfn-python-lint/pull/1789 ?

**Comment by krisb78** [comment]
I think https://github.com/aws-cloudformation/cfn-python-lint/pull/1848 should fix it.

**Comment by kddejong** [comment]
Merged and released with v0.44.2
